### ISSUE TITLE
500 Error After Clicking A Repo Link

### ISSUE THREAD

**Issue by vw98075** [go-gitea/gitea#16725]
- Gitea version (or commit ref):1.15.0+rc3-10-g25437672b
- Git version:Go1.16.5
- Operating system:MacOS
  
  
  
- Database (use `[x]`):
  - [ x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
- Log gist:

```
2021/08/18 17:24:12 Completed GET /user/events 200 OK in 3.363537959s
2021/08/18 17:24:12 Started GET /tk-shoreline/java-code-templete for [::1]:50329
2021/08/18 17:24:12 ...ules/context/repo.go:531:RepoAssignment() [E] RepoAssignment Invalid repo /Users/vewu/Documents/workspace/gitea-1.15/gitea/data/gitea-repositories/tk-shoreline/java-code-templete.git: no such file or directory
2021/08/18 17:24:12 Completed GET /tk-shoreline/java-code-templete 500 Internal Server Error in 8.093535ms
2021/08/18 17:24:12 Started GET /avatar/d6022cd695c8f486ea116a9cd2d17727?size=96 for [::1]:50329
2021/08/18 17:24:12 Completed GET /avatar/d6022cd695c8f486ea116a9cd2d17727?size=96 302 Found in 1.544404ms
```
## Description

Clicking any repo on Gitea will lead to the 500 error. 

This installation is with an existing PostgreSQL DB. I don't know whether that is the cause or not.

## Screenshots

<img width="732" alt="Screen Shot 2021-08-18 at 4 23 21 PM" src="https://user-images.githubusercontent.com/10793038/129988966-57d5f709-8b8d-44f4-9040-0c6dbb88d8d3.png">

**Comment by techknowlogick** [comment]
The error you are getting is `2021/08/18 17:24:12 ...ules/context/repo.go:531:RepoAssignment() [E] RepoAssignment Invalid repo /Users/vewu/Documents/workspace/gitea-1.15/gitea/data/gitea-repositories/tk-shoreline/java-code-templete.git: no such file or directory` are you able to confirm that directory indeed does exist?

**Comment by vw98075** [comment]
I posted the ticket right before I was off. I didn't look error messages in details. I certainly didn't expect that the running time data is stored in a local file system. I don't know how Gitea would run in the cloud if docker containers keep swapping.

**Comment by tacerus** [comment]
After upgrading to v1.15, I have this after clicking any repository link. No error messages can be observed.

**Comment by lunny** [comment]
@tacerus 500 or 404 returend?

**Comment by zeripath** [comment]
> After upgrading to v1.15, I have this after clicking any repository link. No error messages can be observed.

From which version did you upgrade?

Is your `[repository]` `ROOT` correct?

It would be helpful to see some logs.

**Comment by tacerus** [comment]
@lunny 500

@zeripath  Yes, it is correct.
I now downgraded back to 1.14.6 (with a manual database rollback), which sorted the issue for now - unfortunately I forgot to save the trace logs beforehand.
I will try to attempt another upgrade next week and will take care of providing logs then.

Thanks for the quick response!

**Comment by tacerus** [comment]
I was able to try it again sooner.

Here's the log, started just before attempting to open one of the pages (in this case an organization) causing a 500:

https://pasta.lysergic.dev/?6c1b79c0de06b567#7cqeueQrpzupubyJk6nbz5UiAwj4bN1nq7yAHH4Mdx1C

**Comment by zeripath** [comment]
@tacerus 

```
...rs/web/repo/issue.go:1143:ViewIssue() [E] RenderString: RenderContext did not provide context
``` 

This is a duplicate of #16835 and will be fixed by #16842

**Comment by tacerus** [comment]
Thank you!
